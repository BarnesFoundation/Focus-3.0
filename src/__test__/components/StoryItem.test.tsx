import React from "react";
import { unmountComponentAtNode } from "react-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import StoryItem, { StoryItemProps } from "../../components/StoryItem";

let container;
configure({ adapter: new Adapter() }); //enzyme - react 16 hooks support

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("StoryItem", () => {
  it("should match the snapshot", () => {
    const story: StoryItemProps["story"] = {
      detail: {
        art_url: "art_url",
        people: "people",
        title: "title",
        culture: "culture",
        nationality: "nationality",
        birthDate: "birthDate",
        deathDate: "deathDate",
        displayDate: "displayDate",
      },
      long_paragraph: {
        html: "html",
      },
      short_paragraph: {
        html: "html",
      },
    };

    const renderer = new ShallowRenderer();
    renderer.render(
      <StoryItem
        sceneStatus={{ type: "type", state: "state" }}
        statusCallback={(showTitle) => null}
        storyIndex={0}
        onStoryReadComplete={() => null}
        getSize={(scrollOffset, storyIndex) => null}
        storyEmailPage={false}
        isLastStoryItem={false}
        progress={0}
        story={story}
        storyTitle="Story Title"
        onVisChange={(isVisible, storyIndex) => null}
        selectedLanguage={{ code: "En" }}
      />,
      container
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
