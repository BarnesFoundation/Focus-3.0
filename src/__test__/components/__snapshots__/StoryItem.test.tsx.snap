// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryItem should match the snapshot 1`] = `
<div>
  <Timeline
    paused={true}
    totalProgress={0}
  >
    <div
      className="card story-item"
      data-story-card={0}
      data-story-card-last={false}
      id="story-item"
      style={
        Object {
          "transform": "translate3d(0px, -67px, 0px)",
        }
      }
    >
      <div
        className="story-title-bar"
      >
        <div
          className="story-title"
        >
          Story Title
        </div>
      </div>
      <Timeline
        paused={true}
        target={
          <img
            alt="story_item"
            aria-label="title by people used as a background image"
            className="card-img-top"
            role="img"
            src="art_url?crop=faces,entropy&fit=crop&w=0&h=0"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        }
        totalProgress={0}
      >
        <Tween
          duration={0.2}
          ease="easeOut"
          from={
            Object {
              "css": Object {
                "borderRadius": "24px 24px 0px 0px",
              },
            }
          }
          to={
            Object {
              "css": Object {
                "borderRadius": "0px 0px 0px 0px",
              },
            }
          }
        />
      </Timeline>
      <Timeline
        paused={true}
        target={
          <div
            className="overlay"
          />
        }
        totalProgress={0}
      >
        <Tween
          duration={0.1}
          ease="easeOut"
          from={
            Object {
              "autoAlpha": 0,
              "borderRadius": "24px 24px 0px 0px",
            }
          }
          to={
            Object {
              "autoAlpha": 0.6,
              "borderRadius": "0px 0px 0px 0px",
            }
          }
        />
      </Timeline>
      <div
        className="overlay"
        id="overlayRef"
      />
      <div
        className="content-mask"
      >
        <div
          className="card-img-overlay"
        >
          <div
            className="scroll-text"
            id="contentRef"
          >
            <div
              aria-hidden={false}
              className="story-name"
              id="story-here-0"
            >
              Story Title
            </div>
            <VisibilitySensor
              active={true}
              containment={null}
              delayedCall={false}
              intervalCheck={true}
              intervalDelay={100}
              minTopValue={0}
              offset={Object {}}
              onChange={[Function]}
              partialVisibility={false}
              resizeCheck={false}
              resizeDelay={250}
              resizeThrottle={-1}
              scrollCheck={false}
              scrollDelay={250}
              scrollThrottle={-1}
            >
              <div
                className="story-text"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "html",
                  }
                }
                id="story-text-0"
                style={Object {}}
              />
            </VisibilitySensor>
            <div
              className="story-footer"
              style={
                Object {
                  "paddingBottom": "200px",
                }
              }
            >
              title, displayDate
              <br />
              people
										
										 (nationality, birthDate - deathDate)
										
            </div>
          </div>
        </div>
      </div>
    </div>
  </Timeline>
</div>
`;
